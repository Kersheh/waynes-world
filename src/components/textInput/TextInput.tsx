import React from 'react';
import { useFormContext } from 'react-hook-form';

import styles from './TextInput.module.scss';

interface TextInputProps {
  name: string;
  label?: string;
  required?: boolean;
  maxLength?: number;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
const TextInput = ({
  name,
  label,
  required,
  maxLength,
  onChange
}: TextInputProps) => {
  const { watch, register } = useFormContext();

  return (
    <div className={styles.textInput}>
      {label && (
        <div className={styles.label}>
          {label}
          {required && ' *'}
        </div>
      )}
      <input
        type="text"
        name={name}
        required={required}
        maxLength={maxLength}
        ref={register}
        onChange={onChange}
      />
      {maxLength && (
        <div className={styles.counter}>
          {watch()[name]?.length ?? 0} / {maxLength}
        </div>
      )}
    </div>
  );
};

export default TextInput;
