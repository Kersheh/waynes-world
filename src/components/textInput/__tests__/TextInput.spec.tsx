import React from 'react';
import { render } from '@testing-library/react';

import TextInput from '../TextInput';


jest.mock('react-hook-form', () => ({
  useFormContext: () => ({
    watch: jest.fn(),
    register: jest.fn()
  })
}));

describe('<TextInput/> component', () => {
  it('should render', () => {
    const { container } = render(<TextInput name="artist" />);

    expect(container).toMatchSnapshot();
  });
});
