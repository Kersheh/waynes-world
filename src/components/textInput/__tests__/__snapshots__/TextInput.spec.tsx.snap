// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> component should render 1`] = `
<div>
  <div
    class="textInput"
  >
    <input
      name="artist"
      type="text"
    />
  </div>
</div>
`;
